admin/sql/CreateFunctions.sql

```diff
@@ -1928,7 +1928,12 @@ BEGIN
             a_rg.artist,
             -- Withdrawn releases were once official by definition
             bool_and(r.status IS NOT NULL AND r.status != 1 AND r.status != 5),
-            rg.type::SMALLINT,
+            (CASE
+              WHEN rg.type = 3 THEN 2 -- Sort EPs above singles
+              WHEN rg.type = 2 THEN 3 -- Sort singles below EPs
+              ELSE rg.type
+              END
+            )::SMALLINT,
```

Comment:
I'm fine with your solution here as it's pretty straightforward, though I wondered if we should be using `release_group_primary_type.child_order` for this at first. Unless we want the general order of the types in a list to differ from the discography display order? (It would also require a new trigger on `release_group_primary_type` for when the `child_order` changes.)

Reply:
We could use that I guess, yes - it never even crossed my mind since I never think about it at all... Wouldn't that be more annoying though since it'd require new triggers *and* a join here?

Reply:
Tbh I think further order changes are unlikely enough that there might not be a point here - we rarely add primary types and I cannot think of any that would require special sorting.

For *secondary types* that might actually be useful though, if we decide to implement a specific order for those (MBS-13790).

Reply:
> I wondered if we should be using `release_group_primary_type.child_order` for this at first.

Yes, that seems obvious: `child_order` is purposed for ordering, not `type`.

> require new trigger(s) […] taking forever

I’m not sure what trigger(s) you are speaking about.

About search, neither `type` nor `child_order` are looked after, only `gid` and `name`, so there won’t be any issue.

Reply:
> I’m not sure what trigger(s) you are speaking about.


Triggers to update `artist_release_group` whenever `release_group_primary_type` changes, since that means we need that data in here for the sort.